docs/source/overview/cli.md

```diff
@@ -0,0 +1,499 @@
+
+## Operations
+Every file should start with the operations that you want to run. 
+
+```YAML
+ops: [train, evaluate, analyze]
+```
+The `ops` key expects one or more of `[train, evaluate, analyze]` to be specified as a list. In addition to the general & model architecture configurations described in the next 2 sections, each of these operations has an expected set of configurations:
+- `train`: `train_model` (see [Train](#Train)) and `sampler` (see [Samplers used for training](#Samplers-used-for-training))
+- `evaluate`: `evaluate_model` (see [Evaluate](#Evaluate)) and `sampler` (see [Samplers used for evaluation](#Samplers-used-for-evaluate))
+- `analyze`: `analyze_sequences` (see [Analyze](#Analyze)) and one of [`prediction`](#Prediction-on-sequences), [`variant_effect_prediction`](#Variant-effect-prediction), or [`in_silico_mutagenesis`](#In-silico-mutagenesis). 
+
+**Note**: You should be able to use multiple operations (i.e. specify the necessary configuration keys for those operations in a single file). However, if `[train, evaluate]` are both specified, we expect that they will both rely on the same sampler. If you need to train and evaluate using different samplers, please create 2 separate YAML files. 
+
+## General configurations
+In addition to the `ops` key, you can specify the following parameters:
+
+```YAML
+random_seed: 1337
+output_dir: /absolute/path/to/output/dir
+create_subdirectory: True
+lr: 0.01
+```
+
+Note that there should not be any commas at the end of these lines.
+- `random_seed`: Set a random seed for `torch` and `torch.cuda` (if using CUDA) for reproducibility.
+- `output_dir`: The output directory to use for all operations. If no `output_dir` is specified, Selene assumes that the `output_dir` is specified in all relevant function-type values for operations in Selene. (More information on what function-type values are in [later sections](#A-note-for-the-following-sections).)
+- `create_subdirectory`: If True, creates a directory within `output_dir`   with the name formatted as `%Y-%m-%d-%H-%M-%S`---the date/time when Selene was run. (This is only applicable if `output_dir` has been specified.)
+- `lr`:  The learning rate. If you use our [CLI script](https://github.com/FunctionLab/selene/blob/master/selene_cli.py), you can pass this in as a command-line argument rather than having it specified in the configuration file. 
+
+## Model architecture
+For all operations, Selene requires that you specify the model architecture, loss, and optimizer as inputs.
+
+
+### Expected input class and methods
+There are two possible formats you can use to do this:
+
+- single Python file: We expect that most people will start using Selene with model architectures in this format. In this case, you implement your architecture as a class and include 2 static methods, `criterion` and `get_optimizer` in the same file. 
```

Comment:
I found that explanation a little confusing. Perhaps it would be nice to add a link to an example file right here? I think just linking https://github.com/FunctionLab/selene/blob/master/models/deepsea.py might be helpful